### PR TITLE
Added procedure to networkpolicy section

### PR DESCRIPTION
**Pull request by bfallonf**
For: https://trello.com/c/jAKDNfuc/564-document-support-v1-semantics-for-networkpolicy-using-the-existing-object

cc: @danwinship 

Noticed for this card that there's no procedure for the NetworkPolicy section. Asking for ack over email.

### CODE REVIEW

File: admin_guide/managing_networking.adoc [hunk 1]
```diff
@@ -903,3 +903,50 @@ in all other projects. The project to be isolated must first be configured to
...
+spec:
+podSelector:
+ingress:
+- ports:
+  - protocol: TCP
+    - port: 80
+    - port: 443
+----
++
+Optionally, set the `name` parameter to `default-deny` to ensure that the pods that aren't matched to a `NetworkPolicy` continue to be fully-isolated. Delete any existing `NetworkPolicy` objects that previously had the different value. For example, if you are changing the value from `allow-http-and-https` to `default-deny`, delete the old `NetworkPolicy` objects.
```
Comment (danwinship) [inline admin_guide/managing_networking.adoc]: This paragraph is wrong and should be deleted

File: admin_guide/managing_networking.adoc [hunk 2]
```diff
@@ -903,3 +903,50 @@ in all other projects. The project to be isolated must first be configured to
...
+
+. Configure the pod network in the master configuration file at *_/etc/origin/master/master-config.yaml_*:
++
+[source,yaml]
+----
+networkConfig:
+  clusterNetworkCIDR: 10.128.0.0/14 <1>
+  hostSubnetLength: 9 <2>
+  networkPluginName: "redhat/openshift-ovs-networkpolicy" <3>
+  serviceNetworkCIDR: 172.30.0.0/16 <4>
```
Comment (danwinship) [inline admin_guide/managing_networking.adoc]: maybe don't show the `clusterNetworkCIDR` / `hostSubnetLength` / `serviceNetworkCIDR` part and just put `...` instead? To make it clear that all we're really talking about is the `networkPluginName` line. (Likewise below)

File: admin_guide/managing_networking.adoc [hunk 3]
```diff
@@ -903,3 +903,50 @@ in all other projects. The project to be isolated must first be configured to
 opt in to isolation by setting the proper annotation on its Namespace object,
```
Comment (danwinship) [inline admin_guide/managing_networking.adoc]: The paragraph here needs to be updated too... part of it isn't visible in this patch. But in the sentence "network isolation is controlled entirely by `NetworkPolicy` objects and the associated Namespace annotation", you want to remove the words "and the associated Namespace annotation". But keep the external link, just have it on the words "`NetworkPolicy` objects". Then remove the reference to "the proper annotation" in the following sentence as well (which probably means totally rewriting it).

File: admin_guide/managing_networking.adoc (hunk 1 above)
Comment (danwinship) [inline admin_guide/managing_networking.adoc]: To clarify, there's nothing magical about the name "default-deny"; if you want to ensure that all pods in a Namespace are isolated, the way to do that is to create a NetworkPolicy that matches every pod (`spec.podSelector` is empty) but doesn't contain any `ingress` entries. You can call the policy whatever you want, "default-deny" is just a suggestion.

File: admin_guide/managing_networking.adoc [hunk 4]
```diff
@@ -903,3 +903,50 @@ in all other projects. The project to be isolated must first be configured to
...
++
+[source,yaml]
+----
+kind: NetworkPolicy
+apiVersion: extensions/v1beta1
+metadata:
+  name: allow-http-and-https
+spec:
+podSelector:
+ingress:
```
Comment (danwinship) [inline admin_guide/managing_networking.adoc]: `podSelector` and `ingress` should each be indented 2 spaces (and then everything under `ingress` also needs to be indented to match). So basically, everything after `spec` should be indented 2 spaces from where it is now

**Comment by bfallonf** [comment]
Thanks @danwinship . If there's anything else, please let me know.

In the meantime, @openshift/team-documentation this is probably ready for peer review.

File: admin_guide/managing_networking.adoc [hunk 5]
```diff
@@ -898,8 +898,52 @@ using the *ovs-networkpolicy* plug-in.
...
 to use the *ovs-networkpolicy* plug-in], network isolation is controlled
-entirely by `NetworkPolicy` objects and the link:https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network-policy.md[associated Namespace annotation]. In particular, by default, all projects are able to access pods
-in all other projects. The project to be isolated must first be configured to
-opt in to isolation by setting the proper annotation on its Namespace object,
-and then creating `NetworkPolicy` objects indicating the incoming connections to
-be allowed.
+entirely by
+link:https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network-policy.md[`NetworkPolicy`
+objects]. In particular, by default, all projects are able to access pods in all
+other projects. The project to be isolated must first be configured to opt in to
```
Comment (gaurav-nelson) [inline admin_guide/managing_networking.adoc]: 1. Should Namespace object be `Namespace`?

2. Lets change this to active, something like: 

> To isolate a project, opt into isolation by configuring the `Namespace` object, and then create a `NetworkPolicy` object to indicate the allowed incoming connections.

File: admin_guide/managing_networking.adoc [hunk 6]
```diff
@@ -898,8 +898,52 @@ using the *ovs-networkpolicy* plug-in.
...
+. Configure the pod network in the master configuration file at *_/etc/origin/master/master-config.yaml_*:
++
+[source,yaml]
+----
+networkConfig:
+ ...
+  networkPluginName: "redhat/openshift-ovs-networkpolicy" <3>
+ ...
+----
+<1> Cluster network for node IP allocation
```
Comment (gaurav-nelson) [inline admin_guide/managing_networking.adoc]: Code is missing callouts 1,2, and 4

**Comment by gaurav-nelson** [comment]
@bfallonf Done with review, added minor comments.

File: admin_guide/managing_networking.adoc (hunk 6 above)
Comment (bfallonf) [inline admin_guide/managing_networking.adoc]: Oops. Changed one but not the other.

**Comment by bfallonf** [comment]
Thanks @gaurav-nelson . Changed to suggestion. I'll move this along.

**Comment by bfallonf** [comment]
[rev_history]
|xref:../admin_guide/managing_networking.adoc#admin-guide-manage-networking[Managing Networking]
|Added procedure to the xref:../admin_guide/managing_networking.adoc#admin-guide-networking-networkpolicy[Enabling NetworkPolicy] section.
%

File: admin_guide/managing_networking.adoc [hunk 7]
```diff
@@ -898,8 +898,48 @@ using the *ovs-networkpolicy* plug-in.
...
+[source,yaml]
+----
+kind: NetworkPolicy
+apiVersion: extensions/v1beta1
+metadata:
+  name: allow-http-and-https
+spec:
+  podSelector:
+  ingress:
+  - ports:
```
Comment (bmeng) [inline admin_guide/managing_networking.adoc]: the yaml format is incorrect. should be:
```
  - ports:
    - protocol: TCP
      port: 80
    - protocol: TCP
      port: 443
```

File: admin_guide/managing_networking.adoc [hunk 8]
```diff
@@ -898,8 +898,48 @@ using the *ovs-networkpolicy* plug-in.
...
-and then creating `NetworkPolicy` objects indicating the incoming connections to
-be allowed.
+entirely by
+link:https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network-policy.md[`NetworkPolicy`
+objects]. In particular, by default, all projects are able to access pods in all
+other projects. To isolate a project, opt in to isolation by configuring the
+Namespace object, then create a `NetworkPolicy` object to indicate the allowed
+incoming connections.
+
+. Create the `NetworkPolicy` objects:
```
Comment (bmeng) [inline admin_guide/managing_networking.adoc]: The order of the 3 steps looks a little weird. It should be 
1) config master-config.yaml 
2) config node-config.yaml 
3) create the networkpolicy resource 

IMO.

File: admin_guide/managing_networking.adoc [hunk 9]
```diff
@@ -898,8 +898,48 @@ using the *ovs-networkpolicy* plug-in.
...
+      - port: 443
+----
+
+. Configure the pod network in the master configuration file at *_/etc/origin/master/master-config.yaml_*:
++
+[source,yaml]
+----
+networkConfig:
+ ...
+  networkPluginName: "redhat/openshift-ovs-networkpolicy" <3>
```
Comment (bmeng) [inline admin_guide/managing_networking.adoc]: Why the number is `3` here? And the one below is `1`.

Comment (bfallonf) [inline admin_guide/managing_networking.adoc]: My mistake.

**Comment by bfallonf** [comment]
Thanks @bmeng . I have changed as suggested. Please let me know if anything more is needed.

**Comment by bmeng** [comment]
I have not seen the changes in the PR, but I have no other comments beyond those.

**Comment by bfallonf** [comment]
Repushed and verified with @bmeng that the correct info is there. I'll merge.

**Comment by vikram-redhat** [comment]
@bfallonf - there are 2 commits in this PR. Should have been squashed.

**Pull request merged**